src/main/java/com/example/bookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,71 @@
+package com.example.bookstore.controller;
+
+import com.example.bookstore.dto.cart.items.CartItemDto;
+import com.example.bookstore.dto.cart.items.CreateCartItemRequestDto;
+import com.example.bookstore.dto.shopping.cart.CartItemUpdateRequestDto;
+import com.example.bookstore.dto.shopping.cart.ShoppingCartDto;
+import com.example.bookstore.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management",
+        description = "Endpoints for managing user's shopping cart")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @GetMapping
+    @Operation(summary = "Get user's shopping cart")
+    @ResponseStatus(code = HttpStatus.OK)
+    @PreAuthorize("hasAuthority('USER')")
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        return shoppingCartService.getShoppingCartByUserId(authentication);
+    }
+
+    @PostMapping
+    @Operation(summary = "Add cart item to user's shopping cart")
+    @ResponseStatus(code = HttpStatus.CREATED)
```

Comment:
this is wrong response status here

in fact it is corrent only within first request, since then you create a shopping cart

all the subsequent calls operate on existing cart

it will add additional complexity to implement this correctly, so let's keep thing simple and return HTTP 200 here (default response status)